### DISCUSSION TITLE
some question about the best practice of follow squidfunk/mkdocs-material upgrade and my customization

### DISCUSSION THREAD

**Discussion by samzong** [squidfunk/mkdocs-material#4267] [Q&A]
Hi @squidfunk 

I am creating a document project with mkdocs-material, and i love material theme very much, so I hope some way to keep up with the release as much as possible.

but in fact, I don't have the ability to pay for the time being to be a Sponsor. Fortunately, I can complete some development by myself.

I have been looking for a balance between with follow release and customization.

finally, I think maybe use git-submodule mount material into my project, It maybe possible to achieve my goal.

So, I hope to get your suggestions. is this a correct way? if your got better way, looking for to your reply.

THANK U MAKE GREAT TOOLS!

A fans for China

**Comment by squidfunk** [comment]
Thanks for your kind words. The best way to stay close to upstream is by using [theme extension](https://squidfunk.github.io/mkdocs-material/customization/#extending-the-theme). You can override partials or blocks and work with additional JavaScript and CSS to adjust Material for MkDocs to your taste. Templates are kept as stable as possible and usually may only need adjustments between major releases.
